backend/fms_core/viewsets/_utils.py

```diff
@@ -104,30 +111,45 @@ def template_check(self, request):
         if error:
             return HttpResponseBadRequest(json.dumps({"detail": action_data}), content_type="application/json")
 
-        action_def, dataset = action_data
+        action_def, file = action_data
 
         resource_instance = action_def["resource"]()
-        result = resource_instance.import_data(dataset, dry_run=True)
-
-        return Response({
-            "diff_headers": result.diff_headers,
-            "valid": not (result.has_errors() or result.has_validation_errors()),
-            "has_warnings" : any([r.warnings for r in result.rows]),
-            "base_errors": [{
-                "error": str(e.error),
-                "traceback": e.traceback if settings.DEBUG else "",
-            } for e in result.base_errors],
-            "rows": [{
-                "errors": [{
-                    "error": str(e.error),
-                    "traceback": e.traceback if settings.DEBUG else "",
-                } for e in r.errors],
-                "validation_error": r.validation_error,
-                "warnings": r.warnings,
-                "diff": r.diff,
-                "import_type": r.import_type,
-            } for r in result.rows if r.import_type != RowResult.IMPORT_TYPE_SKIP],
-        })
+
+        # result = resource_instance.import_data(dataset, dry_run=True)
+
+        eri = ExperimentRunImporter(file=file, format='xlsx')
```

Comment:
The importer should be called generically (RunImporter) the correct instance should be created using the input.